.github/workflows/ci.yml

```diff
@@ -20,12 +20,16 @@ jobs:
         with:
           fetch-depth: 0 ## Needed for Changesets to find `main` branch
 
+        ## https://github.com/changesets/changesets/issues/517
+      - name: Create git reference to main branch (for Changesets)
+        if: ${{ github.event.pull_request.title != 'Version Packages' }}
+        run: git pull --force --no-tags origin main:main
+
       - uses: ./.github/actions/warm-up-repo
 
       - name: Run yarn lint:changeset
-        if: ${{ success() || failure() }}
+        if: ${{ (success() || failure()) && github.event.pull_request.title != 'Version Packages' }}
         run: |
-          git pull --force --no-tags origin main:main ## https://github.com/changesets/changesets/issues/517
```

Comment:
I moved this command one step up to reduce time gap between repo checkout and branch reference creation